jax/_src/scipy/special.py

```diff
@@ -724,82 +725,249 @@ def i1(x: ArrayLike) -> Array:
-  Wiley-Interscience, 1996.
+YP0 = [
+ 1.55924367855235737965E4,
+-1.46639295903971606143E7,
+ 5.43526477051876500413E9,
+-9.82136065717911466409E11,
+ 8.75906394395366999549E13,
+-3.46628303384729719441E15,
+ 4.42733268572569800351E16,
+-1.84950800436986690637E16,
+]
+YQ0 = [
+ 1.04128353664259848412E3,
+ 6.26107330137134956842E5,
+ 2.68919633393814121987E8,
+ 8.64002487103935000337E10,
+ 2.02979612750105546709E13,
+ 3.17157752842975028269E15,
+ 2.50596256172653059228E17,
+]
 
-  Args:
-    z: The sampling point(s) at which the Bessel function of the first kind are
-      computed.
-    v: The order (int) of the Bessel function.
-    n_iter: The number of iterations required for updating the function
-      values. As a rule of thumb, `n_iter` is the smallest nonnegative integer
-      that satisfies the condition
-      `int(0.5 * log10(6.28 + n_iter) - n_iter *  log10(1.36 + abs(z) / n_iter)) > 20`.
-      Details in `BJNDD` (https://people.sc.fsu.edu/~jburkardt/f77_src/special_functions/special_functions.f)
+DR10 = 5.78318596294678452118E0
+DR20 = 3.04712623436620863991E1
 
-  Returns:
-    An array of shape `(v+1, *z.shape)` containing the values of the Bessel
-    function of orders 0, 1, ..., v. The return type matches the type of `z`.
+RP0 = [
+-4.79443220978201773821E9,
+ 1.95617491946556577543E12,
+-2.49248344360967716204E14,
+ 9.70862251047306323952E15,
+]
+RQ0 = [ 1.0,
+ 4.99563147152651017219E2,
+ 1.73785401676374683123E5,
+ 4.84409658339962045305E7,
+ 1.11855537045356834862E10,
+ 2.11277520115489217587E12,
+ 3.10518229857422583814E14,
+ 3.18121955943204943306E16,
+ 1.71086294081043136091E18,
+]
 
-  Raises:
-    TypeError if `v` is not integer.
-    ValueError if elements of array `z` are not float.
-  """
-  z = jnp.asarray(z)
-  z, = promote_dtypes_inexact(z)
-  z_dtype = lax.dtype(z)
-  if dtypes.issubdtype(z_dtype, complex):
-    raise ValueError("complex input not supported.")
-
-  v = core.concrete_or_error(operator.index, v, 'Argument v of bessel_jn.')
-  n_iter = core.concrete_or_error(int, n_iter, 'Argument n_iter of bessel_jn.')
-
-  bessel_jn_fun = partial(_bessel_jn, v=v, n_iter=n_iter)
-  for _ in range(z.ndim):
-    bessel_jn_fun = vmap(bessel_jn_fun)
-  return jnp.moveaxis(bessel_jn_fun(z), -1, 0)
+def j1_small(x: ArrayLike) -> Array:
+    '''
+    Implementation of J1 for x < 5 
+    '''
+    z = x * x
+    w = jnp.polyval(jnp.array(RP1), z) / jnp.polyval(jnp.array(RQ1), z)
+    w = w * x * (z - jnp.array(Z1)) * (z - jnp.array(Z2))
+    return w
+
+def j1_large(x: ArrayLike) -> Array:  
+    '''
+    Implementation of J1 for x > 5 
+    '''  
+    w = 5.0 / x
+    z = w * w
+    p = jnp.polyval(jnp.array(PP1), z) / jnp.polyval(jnp.array(PQ1), z)
+    q = jnp.polyval(jnp.array(QP1), z) / jnp.polyval(jnp.array(QQ1), z)
+    xn = x - THPIO4
+    p = p * jnp.cos(xn) - w * q * jnp.sin(xn)
+    return p * SQ2OPI / jnp.sqrt(x)
+
+def bessel_j1(z: ArrayLike) -> Array:
```

Comment:
Let's call this `j1`, as it's essentially a JAX implementation of  `scipy.special.j1`

Reply:
Sure thing.